v2/pkg/genruntime/configmaps.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package genruntime
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+)
+
+// ConfigMapReference is a reference to a Kubernetes configmap and key in the same namespace as
+// the resource it is on.
+// +kubebuilder:object:generate=true
+type ConfigMapReference struct {
+	// Name is the name of the Kubernetes configmap being referenced.
+	// The configmap must be in the same namespace as the resource
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Key is the key in the Kubernetes configmap being referenced
+	// +kubebuilder:validation:Required
+	Key string `json:"key"`
+}
+
+// Copy makes an independent copy of the ConfigMapReference
+func (c ConfigMapReference) Copy() ConfigMapReference {
+	return c
+}
+
+func (c ConfigMapReference) String() string {
+	return fmt.Sprintf("Name: %q, Key: %q", c.Name, c.Key)
+}
+
+// ToNamespacedRef creates a NamespacedSecretReference from this SecretReference in the given namespace
+func (c ConfigMapReference) ToNamespacedRef(namespace string) NamespacedConfigMapReference {
+	return NamespacedConfigMapReference{
+		ConfigMapReference: c,
+		Namespace:          namespace,
+	}
+}
+
+// NamespacedConfigMapReference is a ConfigMapReference with namespace information included
+type NamespacedConfigMapReference struct {
+	ConfigMapReference
+	Namespace string
+}
+
+func (s NamespacedConfigMapReference) String() string {
+	return fmt.Sprintf("Namespace: %q, %s", s.Namespace, s.ConfigMapReference.String())
```

Comment:
Do you need the call to `String()` here? I thought `fmt.Sprintf()` would automagically use the `fmt.Stringer` interface if it's present.

Reply:
Fixed